Client.php

```diff
@@ -46,22 +52,47 @@ protected function request($command, $options, $callback)
         $connection = $this->getConnection();
 
         $result = '';
-        $connection->on('data', function($data) use ($result) {
+        $connection->on('data', function($data) use (&$result) {
             $result .= $data;
         });
 
-        $connection->on('close', function() use ($callback, $result) {
+        $connection->on('close', function() use ($callback, &$result) {
```

Comment:
This was my first thought also, but $result as simple type (string) is passed to closure on event registration when it is empty. Without reference when "onClose" is called we will have also empty string as it was "remembered" like this on runtime.

Example:
http://sandbox.onlinephpfunctions.com/code/e8c61ddcab7186d4f95ad52326de651bd1a0d892